app/models/membership_application.rb

```diff
@@ -36,6 +36,7 @@ class MembershipApplication < ApplicationRecord
   accepts_nested_attributes_for :uploaded_files, allow_destroy: true
 
   scope :open, -> { where.not(state: [:accepted, :rejected]) }
+  scope :current, -> { where('id IN (?)', unscoped.select('MAX(id) as id').group(:company_number).map { |application| application.id }) }
```

Comment:
The association between membership_applications (MA) and companies is N-to-M.  

So, a single company can have multiple MA's associated with it.   Each such MA would be for a different user, but the company_number for each MA record would be the same.  

This query just gets the latest MA associated with the company, but leaves other associated MA's behind.

In any case, please see my comment in the "Conversation" tab.  We won't need to perform this kind of scoping for MA's, since we should never have more than one MA - for the same user - for a company.